guava/src/com/google/common/collect/ImmutableEntry.java

```diff
@@ -26,23 +26,23 @@
 @AnnotatedFor({"nullness"})
 @GwtCompatible(serializable = true)
 class ImmutableEntry<K, V> extends AbstractMapEntry<K, V> implements Serializable {
```

Comment:
Do we need to restrict the instantiation of ImmutableEntry to `@NonNull` types?

I haven't added annotations to restrict instantiation of ImmutableEntry to `@NonNull` key-value pairs so as to keep it more flexible. There are certain classes like `com.google.common.collect.HashBiMap.BiEntry` and `com.google.common.
collect.LinkedHashMultimap.ValueEntry` in guava code base itself which extends ImmutableEntry and doesn't necessitate the keys and values to be `@NonNull`. We can always restrict its type to `@NonNull` by explicitly specifying the type during instantiation as it is handled in other `ImmutableXXX` classes. Please let me know if I'm missing something and this class needs to be annotated for null hostility as well. Thanks!

Reply:
That's an interesting fact, and one I would not have expected.  Thanks for discovering it.  It's worth a comment at the declaration of ImmutableEntry.